Index: jcc3/__init__.py
===================================================================
--- jcc3/__init__.py	(revision 1897994)
+++ jcc3/__init__.py	(working copy)
@@ -28,6 +28,13 @@
             path.insert(0, eggpath)
             os.environ['Path'] = os.pathsep.join(path)

+if hasattr(os, 'add_dll_directory'):
+  for location in (('bin', 'client'),('bin', 'server'),('jre', 'bin', 'client'),('jre', 'bin', 'server')):
+    path = os.path.join(os.environ['JAVA_HOME'], *location)
+    if os.path.exists(os.path.join(path, "jvm.dll")):
+      os.add_dll_directory(path)
+      break
+
 import jcc._jcc3 as _jcc3

 # used when jcc is invoked with -m
Index: jcc3/python.py
===================================================================
--- jcc3/python.py	(revision 1897994)
+++ jcc3/python.py	(working copy)
@@ -1621,6 +1621,14 @@
     if shared:
         line(out, 0, "import os, sys")
         line(out)
+
+        line(out, 0, "if hasattr(os, 'add_dll_directory'):")
+        line(out, 1, "for location in (('bin', 'client'),('bin', 'server'),('jre', 'bin', 'client'),('jre', 'bin', 'server')):")
+        line(out, 2, "path = os.path.join(os.environ['JAVA_HOME'], *location)")
+        line(out, 2,"if os.path.exists(os.path.join(path, 'jvm.dll')):")
+        line(out, 3,"os.add_dll_directory(path)")
+        line(out, 3,"break")
+
         line(out, 0, "if sys.platform == 'win32':")
         if find_jvm_dll:
             line(out, 1, "from jcc.windows import add_jvm_dll_directory_to_path")
@@ -1629,6 +1637,14 @@
         line(out, 0, "from . import %s", extname)
     else:
         line(out, 0, "import os")
+
+        line(out, 0, "if hasattr(os, 'add_dll_directory'):")
+        line(out, 1, "for location in (('bin', 'client'),('bin', 'server'),('jre', 'bin', 'client'),('jre', 'bin', 'server')):")
+        line(out, 2, "path = os.path.join(os.environ['JAVA_HOME'], *location)")
+        line(out, 2,"if os.path.exists(os.path.join(path, 'jvm.dll')):")
+        line(out, 3,"os.add_dll_directory(path)")
+        line(out, 3,"break")
+
         line(out, 0, "from . import %s", extname)
     line(out)
     line(out, 0, '__module_dir__ = os.path.abspath(os.path.dirname(__file__))')
