
--- jcc2/python.py	(revision 1809362)
+++ jcc2/python.py	(revision 1809363)
@@ -1691,7 +1691,7 @@
                 else:
                     copytree(module, dst, False)
             else:
-                shutil.copy2(module.split('.')[0] + '.py', modulePath)
+                shutil.copy2(rpartition(module, '.')[0] + '.py', modulePath)
 
     line(out)
     line(out, 0, 'class JavaError(Exception):')


--- jcc3/python.py	(revision 1809362)
+++ jcc3/python.py	(revision 1809363)
@@ -1684,7 +1684,7 @@
                 else:
                     copytree(module, dst, False)
             else:
-                shutil.copy2(module.split('.')[0] + '.py', modulePath)
+                shutil.copy2(module.rpartition('.')[0] + '.py', modulePath)
 
     line(out)
     line(out, 0, 'class JavaError(Exception):')
