--- jcc3/python.py
+++ jcc3/python.py
@@ -1683,7 +1683,7 @@
     line(out, 1, 'def __str__(self):')
     line(out, 2, 'writer = StringWriter()')
     line(out, 2, 'self.getJavaException().printStackTrace(PrintWriter(writer))')
-    line(out, 2, 'return u"\\n".join((unicode(super(JavaError, self)), u"    Java stacktrace:", unicode(writer)))')
+    line(out, 2, 'return u"\\n".join((str(super(JavaError, self)), u"    Java stacktrace:", str(writer)))')
     line(out)
     line(out, 0, 'class InvalidArgsError(Exception):')
     line(out, 1, 'pass')


