Index: jcc3/sources/JCCEnv.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jcc3/sources/JCCEnv.cpp	(revision 1868207)
+++ jcc3/sources/JCCEnv.cpp	(revision 1868208)
@@ -1008,13 +1008,13 @@
           case PyUnicode_1BYTE_KIND: {
               Py_ssize_t len = PyUnicode_GET_LENGTH(object);
               Py_UCS1 *pchars = PyUnicode_1BYTE_DATA(object);
-              std::vector<jchar> jchars;
+              std::vector<jchar> jchars((size_t) len, 0);
 
-              jchars.reserve((size_t) len);
-              for (int i = 0; i < len; ++i)
-                  jchars.push_back((jchar) pchars[i]);
+              for (std::vector<jchar>::iterator it = jchars.begin();
+                   it != jchars.end(); ++it)
+                *it = (jchar) *(pchars++);
 
-              return get_vm_env()->NewString(jchars.data(), jchars.size());
+              return get_vm_env()->NewString(jchars.data(), (size_t) len);
           }
 
           case PyUnicode_2BYTE_KIND: {
